computation/src/main/java/com/powsybl/computation/ComputationException.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.computation;
+
+import com.powsybl.commons.PowsyblException;
+import org.apache.commons.compress.utils.IOUtils;
+
+import javax.annotation.Nullable;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * This class is responsible for carry out the .err/.out logs when execution failed.
+ *
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+public class ComputationException extends PowsyblException {
+
+    private final Map<String, String> outMsgByLogFileName = new HashMap<>();
+
+    private final Map<String, String> errMsgByLogFileName = new HashMap<>();
+
+    private final Map<String, byte[]> zipBytesByFileName = new HashMap<>();
+
+    private final List<Exception> exceptions = new ArrayList<>();
+
+    public ComputationException(String msg) {
+        super(msg);
+    }
+
+    public ComputationException(Throwable throwable) {
+        super(throwable);
+    }
+
+    public ComputationException addOutLog(Path path) {
+        Objects.requireNonNull(path);
+        return addOutLog(path.getFileName().toString(), readFile(path));
+    }
+
+    public ComputationException addErrLog(Path path) {
+        Objects.requireNonNull(path);
+        return addErrLog(path.getFileName().toString(), readFile(path));
+    }
+
+    public ComputationException addFileIfExists(Path path) {
+        if (!Files.exists(path)) {
+            return this;
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
I think Files.readAllBytes(path) will be simpler!

Reply:
Agree